--- a/vendor/oplus/kernel/touchpanel/oplus_touchscreen/Synaptics/Syna_tcm_oncell/synaptics_tcm_oncell.h
+++ b/vendor/oplus/kernel/touchpanel/oplus_touchscreen/Synaptics/Syna_tcm_oncell/synaptics_tcm_oncell.h
@@ -118,6 +118,7 @@
 #define VEE_DETECT      0x05
 #define HEART_DETECT    0x07
 #define UNICODE_DETECT  0x08
+#define STAP_DETECT     0x10
 #define M_UNICODE       0x6d
 #define W_UNICODE       0x77
 
--- a/vendor/oplus/kernel/touchpanel/oplus_touchscreen/Synaptics/Syna_tcm_oncell/synaptics_tcm_oncell.c
+++ b/vendor/oplus/kernel/touchpanel/oplus_touchscreen/Synaptics/Syna_tcm_oncell/synaptics_tcm_oncell.c
@@ -3349,6 +3349,9 @@
         }
         break;
 
+    case STAP_DETECT:
+        gesture->gesture_type = SingleTap;
+        break;
     case TRIANGLE_DETECT:
     default:
         TPD_DEBUG("not support\n");
@@ -3367,6 +3370,11 @@
         gesture->Point_3rd.y    = (touch_data->data_point[18] | (touch_data->data_point[19] << 8));
         gesture->Point_4th.x    = (touch_data->data_point[20] | (touch_data->data_point[21] << 8));
         gesture->Point_4th.y    = (touch_data->data_point[22] | (touch_data->data_point[23] << 8));
+    }
+
+    if (gesture->gesture_type == SingleTap || gesture->gesture_type == DouTap) {
+        gesture->Point_start.x = (touch_data->extra_gesture_info[0] | (touch_data->extra_gesture_info[1] << 8));
+        gesture->Point_start.y = (touch_data->extra_gesture_info[2] | (touch_data->extra_gesture_info[3] << 8));
     }
 
     TPD_INFO("lpwg:0x%x, type:%d, clockwise: %d, points: (%d, %d)(%d, %d)(%d, %d)(%d, %d)(%d, %d)(%d, %d)\n",
